main.py

```diff
@@ -17,14 +17,16 @@ def setup_args():
     """ Setup argparse to take graph name argument. """
     parser = argparse.ArgumentParser(description='Find an Eularian Cicruit.')
     parser.add_argument('graph', nargs='?', help='Name of graph to load')
+    parser.add_argument('start', nargs='?', help='The staring node. Random if none provided.')
```

Comment:
I think you can add `type=int` argument to `add_argument` and eliminate the need for `int(args.start)` later?